script/incremental_build.sh

```diff
@@ -10,10 +10,20 @@ if [ "${BRANCH_NAME}" = "master" ]; then
 else
   # Make sure there is up-to-date master.
   git fetch origin master
-  BRANCH_BASE_SHA=$(git merge-base --fork-point FETCH_HEAD HEAD)
-  echo "Checking changes from $BRANCH_BASE_SHA..."
-  FLUTTER_CHANGED_GLOBAL=`git diff --name-only $BRANCH_BASE_SHA HEAD | grep -v packages | wc -l`
-  FLUTTER_CHANGED_PACKAGES=`git diff --name-only $BRANCH_BASE_SHA HEAD | grep -o "packages/[^/]*" | sed -e "s/packages\///g" | sort | uniq | paste -s -d, -`
+
+  FLUTTER_CHANGED_GLOBAL=0
+  FLUTTER_CHANGED_PACKAGES=""
+
+  # Try get a merge base for the branch and calculate affected packages.
+  # We need this check because some CIs can do a single branch clones with a limited history of commits.
+  if BRANCH_BASE_SHA=$(git merge-base --fork-point FETCH_HEAD HEAD); then
+    echo "Checking changes from $BRANCH_BASE_SHA..."
+    FLUTTER_CHANGED_GLOBAL=`git diff --name-only $BRANCH_BASE_SHA HEAD | grep -v packages | wc -l`
+    FLUTTER_CHANGED_PACKAGES=`git diff --name-only $BRANCH_BASE_SHA HEAD | grep -o "packages/[^/]*" | sed -e "s/packages\///g" | sort | uniq | paste -s -d, -`
+  else
+    echo "Cannot find a merge base for the current branch to run an incremental build!"
```

Comment:
Can you add to this message what would be required to make the build incremental (e.g. rebase, maybe?)?